app/src/pages/models/CloneModelButton.tsx

```diff
@@ -0,0 +1,215 @@
+  onModelCloned?: (model: ModelFormParams) => void;
+  onClose: () => void;
+}) {
+  const data = usePreloadedQuery<CloneModelButtonQuery>(
+    ModelQuery,
+    queryReference
+  );
+  const [commitCloneModel, isCommittingCloneModel] =
+    useMutation<CloneModelButtonMutation>(graphql`
+      mutation CloneModelButtonMutation($input: CreateModelMutationInput!) {
+        createModel(input: $input) {
+          model {
+            id
+            name
+            provider
+            namePattern
+            providerKey
+            tokenCost {
+              input
+              output
+              cacheRead
+              cacheWrite
+              promptAudio
+              completionAudio
+              reasoning
+            }
+          }
+          __typename
+        }
+      }
+    `);
+  const notifySuccess = useNotifySuccess();
+  const notifyError = useNotifyError();
+  const modelData = data?.node;
+
+  if (!modelData) {
+    return <Alert variant="danger">Model not found</Alert>;
+  }
+
+  return (
+    <ModelForm
+      modelName={`${modelData.name} (override)`}
+      modelProvider={modelData.provider}
+      modelNamePattern={modelData.namePattern}
+      modelCost={modelData.tokenCost}
+      onSubmit={(params) => {
+        commitCloneModel({
+          variables: {
+            input: {
+              name: params.name,
+              provider: params.provider,
+              namePattern: params.namePattern,
+              inputCostPerToken: params.cost.input,
+              outputCostPerToken: params.cost.output,
+              cacheReadCostPerToken: params.cost.cacheRead,
+              cacheWriteCostPerToken: params.cost.cacheWrite,
+              promptAudioCostPerToken: params.cost.promptAudio,
+              completionAudioCostPerToken: params.cost.completionAudio,
+              reasoningCostPerToken: params.cost.reasoning,
+            },
+          },
+          onCompleted: () => {
+            onClose();
+            onModelCloned && onModelCloned(params);
+            notifySuccess({
+              title: `Model Cloned`,
+              message: `Model "${params.name}" cloned successfully`,
+            });
+          },
+          onError: (error) => {
+            const formattedError =
+              getErrorMessagesFromRelayMutationError(error);
+            notifyError({
+              title: "Failed to clone model",
+              message: formattedError?.[0] ?? "Failed to clone model",
+            });
+          },
+        });
+      }}
+      isSubmitting={isCommittingCloneModel}
+      submitButtonText="Save Changes"
+      formMode="create"
+    />
+  );
+}
+
+export function CloneModelButton({
+  modelId,
+  onModelCloned,
+}: {
+  modelId: string;
+  onModelCloned?: (model: ModelFormParams) => void;
+}) {
+  const [isOpen, setIsOpen] = useState(false);
+  const [queryReference, loadQuery, disposeQuery] =
+    useQueryLoader<CloneModelButtonQuery>(ModelQuery);
```

Comment:
I think instead of sharing a query variable, it's more idiomatic to declare queries inline and then share named fragments when it makes sense.